daemon/src/routes_maker.rs

```diff
@@ -122,27 +122,45 @@ pub struct PromptAuthentication {
     www_authenticate: Header<'static>,
 }
 
-// // TODO: Shall we use a simpler struct for verification? AFAICT quantity is not
-// // needed, no need to send the whole CFD either as the other fields can be generated from the
-// order #[rocket::post("/order/confirm", data = "<cfd_confirm_order_request>")]
-// pub async fn post_confirm_order(
-//     cfd_confirm_order_request: Json<CfdTakeRequest>,
-//     queue: &State<mpsc::Sender<CfdOrder>>,
-//     mut conn: Connection<Db>,
-// ) -> Result<status::Accepted<()>, status::BadRequest<String>> {
-//     dbg!(&cfd_confirm_order_request);
-
-//     let order = db::load_order_by_id_from_conn(cfd_confirm_order_request.order_id, &mut conn)
-//         .await
-//         .map_err(|e| status::BadRequest(Some(e.to_string())))?;
-
-//     let _res = queue
-//         .send(order)
-//         .await
-//         .map_err(|_| status::BadRequest(Some("internal server error".to_string())))?;
-
-//     Ok(status::Accepted(None))
-// }
+/// The maker POSTs this to accept an order
+#[derive(Debug, Clone, Deserialize)]
+pub struct AcceptOrRejectOrderRequest {
+    pub taker_id: TakerId,
```

Comment:
Why do we need to send the `TakerId` along here? We should be able to know from the `OrderId`, where we need to send the information yes?

Having to compare that this `TakerId` is indeed the one that we have stored internally looks like an unnecessary error case.

Reply:
There is a potential racecondition here. I opted for keeping the taker id in this request so that we know which order we are accepting. 

I shortly summarized it in here: https://github.com/comit-network/hermes/issues/120

Reply:
We can only have a race condition if we don't update the state properly. Actors process messages in sequence. If the state update within an incoming message like `TakeOrder` is sound, there can be no race conditions.

At the moment, we remove the currently active order as soon as it is taken. With your PR, we update the state to `IncomingOrderRequest` upon `TakeOrder`.

The only thing to prevent a race condition is to check that the order is not already in that state when another `TakeOrder` comes in.

Reply:
I'll remove it for now as it's not really consistent. The ticket remains though and we should takle it sooner or later.